tests/testthat/test-geom-violin.R

```diff
@@ -46,6 +46,48 @@ test_that("quantiles do not fail on zero-range data", {
   expect_equal(length(layer_grob(p)), 1)
 })
 
+test_that("quantiles are at expected positions", {
+  get_quantile_line <- function(y) {
+    df <- data.frame(x = 1, y = y)
+
+    p <- ggplot(df, aes(x = x, y = y)) +
+      geom_violin(draw_quantiles = c(0.5), bw = 0.1)
+
+    grob <- layer_grob(p)
```

Comment:
Instead of testing the grobs, could you just test `create_quantile_segment_frame()` directly?

Reply:
Good idea, done. In this process, I removed the test for the non-zero middle because that is covered by "`create_quantile_segment_frame` functions for 3 quantiles".